p2p/pex/pex_reactor.go

```diff
@@ -216,11 +216,11 @@ func (r *PEXReactor) logErrAddrBook(err error) {
 	}
 }
 
-// RemovePeer implements Reactor.
-func (r *PEXReactor) RemovePeer(p Peer, reason interface{}) {
-	id := string(p.ID())
+func (r *PEXReactor) InitPeer(peer Peer) Peer {
+	id := string(peer.ID())
 	r.requestsSent.Delete(id)
 	r.lastReceivedRequests.Delete(id)
+	return peer
```

Comment:
do we need to remove these lines from `RemovePeer`? **if no**, we should extract them to a function call it in both places